### PR TITLE
Change to OpenAI embedding API

### LINKED ISSUE #2296: OpenAI module use Embedding API instead of deprecated Models API
Link to currently used API: https://beta.openai.com/docs/api-reference/engines

Link to new API that should be used: https://beta.openai.com/docs/api-reference/embeddings/create

### PR DESCRIPTION
**Pull request by samos123**
The previous engines API was deprecated and so changing to embedding API was required

This fixes #2296

Tested by running openAI demo here:
https://github.com/semi-technologies/DEMO-text2vec-openai

### Review checklist

- [x] Documentation has been updated, if necessary. Link to changed documentation:
No end-user impact so no documentation change needed

- [ ] Chaos pipeline run or not necessary. Link to pipeline:
Unclear on this one

- [x] All new code is covered by tests where it is reasonable.
Same amount of coverage as before

- [ ] Performance tests have been run or not necessary.
Have not been run

### CODE REVIEW

**Comment by samos123** [comment]
@antas-marcin Please review

**Comment by samos123** [comment]
I agree

File: modules/text2vec-openai/clients/vectorizer.go [hunk 1]
```diff
@@ -50,41 +51,46 @@ type openAIApiError struct {
...
 ) (*ent.VectorizationResult, error) {
 	body, err := json.Marshal(embeddingsRequest{
 		Input: input,
+		Model: model,
 	})
 	if err != nil {
 		return nil, errors.Wrapf(err, "marshal body")
 	}
 
+	url := v.host + v.path
```
Comment (antas-marcin) [inline modules/text2vec-openai/clients/vectorizer.go:92]: just a suggestion, you could use `url.JoinPath()` method here

File: modules/text2vec-openai/clients/vectorizer_test.go [hunk 2]
```diff
@@ -32,15 +32,10 @@ func TestClient(t *testing.T) {
...
-			apiKey:     "apiKey",
-			httpClient: &http.Client{},
-			urlBuilder: &openAIUrlBuilder{
-				origin:   server.URL,
-				pathMask: "/v1/engines/%s-search-%s-%s-001/embeddings",
-			},
-			logger: nullLogger(),
-		}
+
+		c := New("apiKEy", nullLogger())
```
Comment (antas-marcin) [inline modules/text2vec-openai/clients/vectorizer_test.go:36]: I think that here it should be `apiKey` not `apiKEy`

File: modules/text2vec-openai/clients/vectorizer_test.go [hunk 3]
```diff
@@ -59,15 +54,8 @@ func TestClient(t *testing.T) {
...
-		c := &vectorizer{
-			apiKey:     "apiKey",
-			httpClient: &http.Client{},
-			urlBuilder: &openAIUrlBuilder{
-				origin:   server.URL,
-				pathMask: "/v1/engines/%s-search-%s-%s-001/embeddings",
-			},
-			logger: nullLogger(),
-		}
+		c := New("apiKEy", nullLogger())
```
Comment (antas-marcin) [inline modules/text2vec-openai/clients/vectorizer_test.go:57]: `apiKey`

File: modules/text2vec-openai/clients/vectorizer_test.go [hunk 4]
```diff
@@ -83,15 +71,8 @@ func TestClient(t *testing.T) {
...
```
Comment (antas-marcin) [inline modules/text2vec-openai/clients/vectorizer_test.go:74]: `apiKey`

File: modules/text2vec-openai/clients/vectorizer.go (hunk 1 above)
Comment (samos123) [reply modules/text2vec-openai/clients/vectorizer.go:92]: confirmed by @antas-marcin that it's fine to require 1.19

File: modules/text2vec-openai/clients/vectorizer_test.go (hunk 4 above)
Comment (samos123) [reply modules/text2vec-openai/clients/vectorizer_test.go]: done

File: modules/text2vec-openai/clients/vectorizer_test.go (hunk 3 above)
Comment (samos123) [reply modules/text2vec-openai/clients/vectorizer_test.go]: done

File: modules/text2vec-openai/clients/vectorizer.go (hunk 1 above)
Comment (samos123) [reply modules/text2vec-openai/clients/vectorizer.go]: done

**Pull request merged**